ballerina/graphql_client.bal

```diff
@@ -0,0 +1,107 @@
+    # + headers - The GraphQL API headers to execute each query 
+    # + targetType - The payload, which is expected to be returned after data binding. For example 
+    #                `type CountryByCodeResponse record {| map<json?> __extensions?; record {| record{|string name;|}? country; |} data;`
+    # + return - The payload (if the `targetType` is configured) or a `graphql:ClientError` if failed to execute the query
+    remote isolated function executeWithType(string query, map<anydata>? variables = (), 
+                                             map<string|string[]>? headers = (), 
+                                             typedesc<GenericResponse|record{}|json> targetType = <>) 
+                                             returns targetType|ClientError = @java:Method {
+        'class: "io.ballerina.stdlib.graphql.runtime.client.QueryExecutor",
+        name: "executeWithType"
+    } external;
+
+    private isolated function processExecuteWithType(typedesc<GenericResponse|record{}|json> targetType, 
+                                                     string query, map<anydata>? variables, 
+                                                     map<string|string[]>? headers) 
+                                                     returns GenericResponse|record{}|json|ClientError {
+        http:Request request = new;
+        json graphqlPayload = getGraphqlPayload(query, variables);
+        request.setPayload(graphqlPayload);
+        json|http:ClientError httpResponse = self.httpClient->post("", request, headers = headers);
+
+        if httpResponse is http:ClientError {
+            check handleHttpClientErrorResponse(httpResponse);
+        } else {
+            map<json> responseMap = <map<json>> httpResponse;
+            if responseMap.hasKey("errors") {
+                check handleGraphqlErrorResponse(responseMap);
+            } else {
+                return check performDataBinding(targetType, responseMap);
+            }
+        }
+    }
+
+    # Executes a GraphQL query and data binds the GraphQL response to a record with data, extensions and errors 
+    # which is a subtype of GenericResponseWithErrors. The errors of the GraphQL response are directed in the Ballerina 
+    # target type data binding path.
+    #
+    # + query - The GraphQL query. For example `query countryByCode($code:ID!) {country(code:$code) {name}}`.
+    # + variables - The GraphQL variables. For example `{"code": "<variable_value>"}`.
+    # + headers - The GraphQL API headers to execute each query 
+    # + targetType - The payload (`GenericResponseWithErrors`), which is expected to be returned after data binding. For example 
+    #               `type CountryByCodeResponse record {| map<json?> __extensions?; record {| record{|string name;|}? country; |} data; ErrorDetail[] errors?; |};`
+    # + return - The payload (if the `targetType` is configured) or a `graphql:ClientError` if failed to execute the query
```

Comment:
The word `payload` is not a GraphQL word. It should be the GraphQL response.

```suggestion
    # + return - The payload or a `graphql:ClientError` if failed to execute the query
```

Reply:
The change is applied in [47e5a38](https://github.com/ballerina-platform/module-ballerina-graphql/pull/663/commits/47e5a38537e0e31ee452d51e9943f81c773e3a95). Thank you.